lightning_client.go

```diff
@@ -2703,3 +2708,60 @@ func (s *lightningClient) NetworkInfo(ctx context.Context) (*NetworkInfo,
 		NumZombieChans:       resp.NumZombieChans,
 	}, nil
 }
+
+// InvoiceSubscription holds the parameters to specify from which update
+// to start streaming
+type InvoiceSubscription struct {
+	//
+	//If specified (non-zero), then we'll first start by sending out
+	//notifications for all added indexes with an add_index greater than this
+	//value. This allows callers to catch up on any events they missed while they
+	//weren't connected to the streaming RPC.
+	AddIndex uint64
+	//
+	//If specified (non-zero), then we'll first start by sending out
+	//notifications for all settled indexes with an settle_index greater than
+	//this value. This allows callers to catch up on any events they missed while
+	//they weren't connected to the streaming RPC.
+	SettleIndex uint64
+}
+
+// SubscribeInvoices allows a client to subscribe to updates
+// of newly added/settled invoices.
+func (s *lightningClient) SubscribeInvoices(ctx context.Context, req InvoiceSubscription) (<-chan *lnrpc.Invoice, <-chan error, error) {
+	rpcCtx, cancel := context.WithTimeout(ctx, rpcTimeout)
+	defer cancel()
```

Comment:
Can't use a cancel and/or timeout here, otherwise the subscription will be killed as soon as this method returns. See `SubscribeChannelEvents` for example.

Reply:
You are totally right, I've fixed it

Reply:
I understand why It is wrong. I've fixed it.